src/app.js

```diff
@@ -16,6 +18,10 @@ const state = {
 const app = remote.app;
 const appDir = jetpack.cwd(app.getAppPath());
 
+if (IS_LINUX) {
+  app.mainWindow.setIcon(path.join(__dirname, '..', 'resources', 'icons', '128x128.png'))
+}
```

Comment:
Probably worth checking if this has any unwanted effects on other Linux distros.  
I don't think I'll have the chance to check personally anytime soon, but if I do I'll add a comment here.

Reply:
Definitely, though I'm surprised it didn't work out of the box on Arch because it seems to work on Ubuntu with GNOME.